include/revng/PipelineC/Prototypes.h

```diff
@@ -152,6 +152,17 @@ bool rp_manager_set_global(rp_manager *manager,
                            const char *serialized,
                            const char *global_name);
 
+/**
+ * \returns the number of serializable global objects
+ */
+int rp_manager_get_globals_count(rp_manager *manager);
+
+/**
+ * \returns the owning pointer to the name of serializable global object
+ * with the provided index, nullptr if the index was out of bound.
+ */
+const char *rp_manager_create_global_name(rp_manager *manager, int index);
```

Comment:
* Why the `index`?
* What is this function for?

Reply:
now we can have multiple model like globals, we need to fetch a key for them, in this case the name

Reply:
Aaaah, so this is `rp_manger_get_global_name`, but you put `_create` due to the fact it returns an owning pointer.
Why does it have to be owning? Can't we state that its lifetime is associated to `rp_manager`'s?
Isn't this the case for other strings too?

Reply:
it was just because the underlying structure is a string map and there was no guarantee of being zero terminated